Content.Server/Administration/ServerAPI.cs

```diff
@@ -0,0 +1,795 @@
+    {
+        if (context.RequestMethod != HttpMethod.Post || context.Url!.AbsolutePath != "/admin/actions/add_game_rule")
+        {
+            return false;
+        }
+
+        if (!CheckAccess(context))
+        {
+            await context.RespondErrorAsync(HttpStatusCode.Unauthorized);
+            return true;
+        }
+
+        if (!CheckActor(context, out var actor))
+        {
+            await context.RespondAsync("An actor is required to perform this action.", HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        var gameRuleSupplied = context.RequestHeaders.TryGetValue("GameRuleId", out var gameRule);
+        if (!gameRuleSupplied)
+        {
+            _sawmill.Info($"Attempted to add game rule without supplying a game rule name by {actor!.Name}({actor!.Guid}).");
+            await context.RespondErrorAsync(HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        var ticker = _entitySystemManager.GetEntitySystem<GameTicker>();
+
+        var tsc = new TaskCompletionSource<EntityUid>();
+        _taskManager.RunOnMainThread(() =>
+        {
+            var ruleEntity = ticker.AddGameRule(gameRule.ToString());
+            _sawmill.Info($"Added game rule {gameRule} by {actor!.Name}({actor!.Guid}).");
+            if (ticker.RunLevel == GameRunLevel.InRound)
+            {
+                ticker.StartGameRule(ruleEntity);
+                _sawmill.Info($"Started game rule {gameRule} by {actor!.Name}({actor!.Guid}).");
+            }
+            tsc.TrySetResult(ruleEntity);
+        });
+
+        var ruleEntity = await tsc.Task;
+        await context.RespondAsync($"Added game rule {ruleEntity}", HttpStatusCode.OK);
+        return true;
+    }
+
+    /// <summary>
+    ///     Kicks a player.
+    /// </summary>
+    private async Task<bool> ActionKick(IStatusHandlerContext context)
+    {
+        if (context.RequestMethod != HttpMethod.Post || context.Url!.AbsolutePath != "/admin/actions/kick")
+        {
+            return false;
+        }
+
+        if (!CheckAccess(context))
+        {
+            await context.RespondErrorAsync(HttpStatusCode.Unauthorized);
+            return true;
+        }
+
+        if (!CheckActor(context, out var actor))
+        {
+            await context.RespondAsync("An actor is required to perform this action.", HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        var playerSupplied = context.RequestHeaders.TryGetValue("Guid", out var guid);
+        if (!playerSupplied)
+        {
+            _sawmill.Info($"Attempted to kick player without supplying a username by {actor!.Name}({actor!.Guid}).");
+            await context.RespondErrorAsync(HttpStatusCode.BadRequest);
+            return true;
+        }
+
+        var session = await RunOnMainThread(() =>
+        {
+            // There is no function to get a session by GUID, so we have to iterate over all sessions and check their GUIDs.
```

Comment:
`IPlayerManager.TryGetSessionById`

Reply:
😭 I will never recover from this

Reply:
`IPlayerManager.TryGetSessionById` takes in a NetUserId, I only have the GUID. I cannot use that function

Reply:
`new NetUserId(guid)`

Reply:
My career is ruined...